tests/unit/render/test_render.cc

```diff
@@ -0,0 +1,199 @@
+
+#include <string>
+#include <filesystem>
+#include <iostream>
+#include <variant>
+#include <set>
+#include <regex>
+
+#include "../generator.h"
+
+using namespace vt::tv::tests::unit;
+
+namespace vt::tv::tests::unit::render {
+
+/**
+ * Provides unit tests for the vt::tv::render::Render class to test with config file input
+ */
+class RenderTest :public ::testing::TestWithParam<std::string> {
+
+  void SetUp() override {
+    // Disable this test because of gcc segfault at vtkWindowToImageFilter (memcpy) if save png is true in config
+    // GTEST_SKIP();
+    // return;
+
+    // Make the output directory for these tests
+    std::filesystem::create_directory(fmt::format("{}/output", SRC_DIR));
+    std::filesystem::create_directory(fmt::format("{}/output/tests", SRC_DIR));
+    std::filesystem::create_directory(fmt::format("{}/output/tests/render", SRC_DIR));
+  }
+
+  protected:
+    Render createRender(YAML::Node config, Info info, std::string &output_dir) {
+      // change output directory to use specific one for these tests
+      output_dir = "output/tests/render";
+
+      std::filesystem::path output_path(output_dir);
+      if (output_path.is_relative()) {
+        output_path = std::filesystem::path(SRC_DIR) / output_path;
+      }
+      output_dir = output_path.string();
+
+      // append / to avoid problems with file stems
+      if (!output_dir.empty() && output_dir.back() != '/') {
+        output_dir += '/';
+      }
+
+      return Render(
+      {
+          config["viz"]["rank_qoi"].as<std::string>(),
+          "",
+          config["viz"]["object_qoi"].as<std::string>()
+        },
+        config["viz"]["force_continuous_object_qoi"].as<bool>(),
+        info,
+        {
+          config["viz"]["x_ranks"].as<uint64_t>(),
+          config["viz"]["y_ranks"].as<uint64_t>(),
+          config["viz"]["z_ranks"].as<uint64_t>()
+        },
+        config["viz"]["object_jitter"].as<double>(),
+        output_dir,
+        config["output"]["file_stem"].as<std::string>(),
+        1.0,
+        config["viz"]["save_meshes"].as<bool>(),
+        // config["viz"]["save_pngs"]: Let PNG rendering to the ParseRenderTest test class.
+        false,
+        std::numeric_limits<PhaseType>::max()
+      );
+    }
+};
+
+/**
+ * Test Render:generate correcty run the different configuration files
+ */
+TEST_P(RenderTest, test_render_from_config) {
+  std::string const & config_file = GetParam();
+  YAML::Node config = YAML::LoadFile(fmt::format("{}/tests/config/{}", SRC_DIR, config_file));
+  Info info = Generator::loadInfoFromConfig(config);
+
+  uint64_t win_size = 2000;
+  uint64_t font_size = 50;
+
+  std::string output_dir;
+  if (config["viz"]["object_qoi"].as<std::string>() == "shared_block_id") {
+    // Temporary: this case must be removed as soon as the `shared_block_id` QOI becomes supported.
+    EXPECT_THROW(createRender(config, info, output_dir), std::runtime_error); // "Invalid Object QOI: shared_block_id"
```

Comment:
`shared_block_id` should be supported via the `attributes` feature of vttv (see[ this comment](https://github.com/DARMA-tasking/vt-tv/issues/80#issuecomment-2233686338))

Reply:
attribute value is restored. Faling in CI for the moment